R/parallelMap.R

```diff
@@ -108,14 +110,29 @@ parallelMap = function(fun, ..., more.args = list(), simplify = FALSE, use.names
 
     if (isModeMulticore()) {
       more.args = c(list(.fun = fun, .logdir = logdir), more.args)
-      res = MulticoreClusterMap(slaveWrapper, ..., .i = iters, MoreArgs = more.args, mc.cores = cpus,
+      if (reproducible) {
+        old.seed = .Random.seed
+        seed = sample(1:100000, 1)
+        # we need to reset the seed first in case the user supplied a seed,
+        # otherwise "L'Ecuyer-CMRG" won't be used
+        rm(.Random.seed, envir = globalenv())
+        set.seed(seed, "L'Ecuyer-CMRG")
+      }
+      res = MulticoreClusterMap(slaveWrapper, ..., .i = iters,
+        MoreArgs = more.args, mc.cores = cpus,
         SIMPLIFY = FALSE, USE.NAMES = FALSE)
+      if (reproducible) {
+        .Random.seed = old.seed
+        RNGkind("Mersenne-Twister")
```

Comment:
It could have been something else then Mersenne-Twister before, right?

Reply:
Indeed! I've changed it to restore any initial RNG kind with all of its options.